HWs/HW6/Game/Game/Game.cs

```diff
@@ -0,0 +1,272 @@
+namespace Game;
+
+using static System.Console;
+using System.Text;
+
+/// <summary>
+/// The class responsible for the mechanics of the game.
+/// </summary>
+public class Game
+{
+    /// <summary>
+    /// Draws a map from a file, searches for and puts the player at the starting point.
+    /// </summary>
+    /// <param name="filePath"> The File path.</param>
+    public Game(string filePath)
+    {
+        string[] storage;
+        try
+        {
+            storage = File.ReadAllLines(filePath);
+            if (storage.Count() == 0 || storage[0].Count() == 0)
+            {
+                throw new ArgumentException("File empty.");
+            }
+        }
+        catch (Exception e) when (e is ArgumentException || e is ArgumentNullException
+                                  || e is IOException || e is DirectoryNotFoundException)
+        {
+            throw;
+        }
+
+        for (int i = 0; i < storage.Count(); ++i)
+        {
+            _map.Add(new StringBuilder(storage[i]));
+        }
+
+        this._player = new Player(FindTheStartingPoint());
+
+        if (_player.Coordinates == (0, 0))
+        {
+            throw new ArgumentException("Inccorect map.");
+        }
+
+        DrawTheMap();
+    }
+
+    /// <summary>
+    /// Method of obtaining the current state of the map and the player.
+    /// </summary>
+    /// <returns> Array of rows, where the row is one horizontal line of the map. </returns>
+    public string[] GetMapAndPlayer()
+    {
+        _map[_player.Coordinates.Y][_player.Coordinates.X] = '@';
+        var result = new string[_map.Count + 1];
+        for (int i = 0; i < _map.Count; i++)
+        {
+            result[i] = _map[i].ToString();
+        }
+        result[_map.Count] = $"Health: {_player.Health}";
+        _map[_player.Coordinates.Y][_player.Coordinates.X] = ' ';
+        return result;
+    }
+
+    private List<StringBuilder> _map = new List<StringBuilder>();
+
+    private Player _player;
+
+    /// <summary>
+    /// Enumerating directions.
+    /// </summary>
+    public enum Direction
+    {
+        Left,
+        Right,
+        Up,
+        Down,
+    }
+    private const int _treatmentPoints = 10;
+
```

Comment:
```suggestion
    private const int _criticalHealth = _damagePoints * 3;
```
Константы можно определять с помощь. других констант. Так и эффективнее (константа гарантированно вычисляется лишь один раз), и изменение кода проще (изменить это значение нужно только в одном месте). И код читаемее - именованная константа вместо волшебных выражений вида
```
this._player.Health <= _damagePoints * 3
```